File: OOPS/Python/Inheritance.py
class Vehicle:
    def __init__(self,registraion_no,no_of_wheels,color):
        self.registration_no=registartion_no
        self.no_of_wheels=no_of_wheels
        self.color=color
    def getDetails(self):
        print("Registration number is ",self.registration_no)
        print("Number of wheels is ",self.no_of_wheels)
        print("Color is ",color)
class Fourwheeler(Vehicle):
    def __init__(self, registraion_no, no_of_wheels, color,torque,gear_box_type):
        super().__init__(registraion_no, no_of_wheels, color)
        self.torque=torque
        self.gear_box_type=gear_box_type
    def getDetails(self):
        super().getDetails()
        print("Torque is ",self.torque)
        print("Gear box type is ",self.gear_box_type)
if __name__=="__main__":
    f1=Fourwheeler(1234,4,"gray",345,"automatic")
    f1.getDetails()

File: OOPS/Inheritance.java
// class Vehicle {
//     int registration_no;
//     int no_of_wheels;
//     String color;
//     Vehicle(int registration_no,int no_of_wheels,String color){
//         this.registration_no=registration_no;
//         this.no_of_wheels=no_of_wheels;
//         this.color=color;
//     }
//     void getDetails(){
//         System.out.println("Registration number is "+this.registration_no);
//         System.out.println("Number of wheels is "+this.no_of_wheels);
//         System.out.println("Color is "+this.color);
//     }
// }
// class Fourwheeler extends Vehicle{
//     int torque;
//     String gear_box_type;
//     Fourwheeler(int registration_no,int no_of_wheels,String color,int torque,String gear_box_type){
//         super(registration_no, no_of_wheels, color);
//         this.gear_box_type=gear_box_type;
//         this.torque=torque;
//     }
//     void getDetails(){
//         super.getDetails();
//         System.out.println("Torque is "+this.torque);
//         System.out.println("Gear box type is  "+this.gear_box_type);
//     }
// }
// public class Inheritance{
//     public static void main(String [] args){
//         Fourwheeler f1=new Fourwheeler(1234, 4,"gray", 345, "automatic");
//         f1.getDetails();
//     }
// }